src/envoy/utils/utils_test.cc

```diff
@@ -69,4 +121,45 @@ TEST(UtilsTest, CheckResponseInfoToStreamInfo) {
   CheckResponseInfoToStreamInfo(check_response, mock_stream_info);
 }
 
+TEST_P(UtilsTest, GetPrincipal) {
+  std::vector<std::string> sans{"spiffe://foo/bar", "bad"};
+  testGetPrincipal(sans, "foo/bar", true);
+}
+
+TEST_P(UtilsTest, GetPrincipalNoSpiffePrefix) {
+  std::vector<std::string> sans{"spiffe:foo/bar", "bad"};
+  testGetPrincipal(sans, "spiffe:foo/bar", true);
+}
+
+TEST_P(UtilsTest, GetPrincipalEmpty) {
+  std::vector<std::string> sans;
+  testGetPrincipal(sans, "", false);
+}
+
+TEST_P(UtilsTest, GetTrustDomain) {
+  std::vector<std::string> sans{"spiffe://td/bar", "bad"};
```

Comment:
What do you mean by "bad" here? My understanding ist hat getTrustDomain use only the first element. So the second element is the hint whether the test should successful? If so it is not "bad"

Reply:
Ah, I just put it here as a second element to make sure the 1st element is used. It could be anything as long as it's different from the 1st element.